src/modules/esl-event-listener/core/targets/swipe.target.ts

```diff
@@ -0,0 +1,167 @@
+    // if the user released on a different target, cancel!
+    if (!this.isGestureStarted || (this.startEvent?.target !== e.target)) return;
+
+    const eventsDiff = this.eventDiff(e);
+    const direction = this.resolveDirection(eventsDiff);
+    if (direction) {
+      const swipeInfo: ESLSwipeGestureEventInfo = {
+        target: this.target,
+        direction,
+        distanceX: Math.abs(eventsDiff.x),
+        distanceY: Math.abs(eventsDiff.y),
+        startEvent: this.startEvent,
+        endEvent: e
+      };
+
+      // fire `swipe` event on the element that started the swipe
+      this.dispatchEvent(ESLSwipeGestureEvent.fromConfig('swipe', swipeInfo));
+      // fire `swipe:${dir}` event on the element that started the swipe
+      this.dispatchEvent(ESLSwipeGestureEvent.fromConfig(`swipe:${direction}` as SwipeEventName, swipeInfo));
+    }
+
+    // Mark swipe as completed
+    this.isGestureStarted = false;
+  }
+
+  /**
+   * Returns swipe direction based on distance between swipe start and end points
+   * @param diff - diff between pointerdown and pointer coordinates and timestamp {@link EventsDiff}
+   * @returns direction of swipe {@link SwipeDirection}
+   */
+  protected resolveDirection(diff: EventsDiff): SwipeDirection | null {
+    const swipeThreshold = (resolveCSSSize(this.config.threshold) || resolveCSSSize(ESLSwipeGestureTarget.defaultConfig.threshold)!);
+
+    if (Math.abs(diff.x) > Math.abs(diff.y) && Math.abs(diff.x) > swipeThreshold && diff.time < this.config.timeout) {
+      return diff.x > 0 ? 'left' : 'right';
+    }
+    if (Math.abs(diff.y) > swipeThreshold && diff.time < this.config.timeout) {
+      return diff.y > 0 ? 'up' : 'down';
+    }
+
+    // Marker that there was not enough move characteristic to consider pointer move as touch swipe
+    return null;
+  }
+
+  /**
+   * Subscribes to pointerup and pointerdown event
+   */
+  public override addEventListener(callback: EventListener): void;
+  public override addEventListener(event: SwipeEventName, callback: EventListener): void;
+  public override addEventListener(event: any, callback: EventListener = event): void {
+    super.addEventListener(event, callback);
+    if (this.getEventListeners().length > 1) return;
+
+    ESLEventUtils.subscribe(this.target, {event: 'pointerdown', capture: false}, this.handleStart);
+    ESLEventUtils.subscribe(this.target, {event: 'pointerup', capture: false}, this.handleEnd);
+  }
+
+  /**
+   * Unsubscribes from the observed target {@link Element} changes
+   */
+  public override removeEventListener(callback: EventListener): void;
+  public override removeEventListener(event: SwipeEventName, callback: EventListener): void;
+  public override removeEventListener(event: any, callback: EventListener = event): void {
+    super.removeEventListener(event, callback);
+    if (this.getEventListeners().length > 0) return;
+
+    ESLEventUtils.unsubscribe(this.target, 'pointerdown');
```

Comment:
I guess, here we should specify handler
ESLEventUtils.unsubscribe(this.target, this.handleStart);
ESLEventUtils.unsubscribe(this.target, this.handleEnd);